discord/bot.py

```diff
@@ -576,8 +576,8 @@ async def sync_commands(
             guild_permissions: List = []
 
             for i in commands:
-                cmd = find(lambda cmd: cmd.name == i["name"] and cmd.type == i["type"] and cmd.guild_ids is not None
-                                       and (i["guild_id"]) in cmd.guild_ids, self.pending_application_commands)
+                cmd = find(lambda cmd: cmd.name == i["name"] and str(cmd.type) == str(i["type"]) and cmd.guild_ids is not None
+                                       and (int(i["guild_id"])) in cmd.guild_ids, self.pending_application_commands)
```

Comment:
```suggestion
                                       and int(i["guild_id"]) in cmd.guild_ids, self.pending_application_commands)
```

Remove unnecessary parenthesis - this was actually a missing int() per here: https://discord.com/channels/881207955029110855/881735314987708456/939338589412655145